### ISSUE TITLE
Gyroid infill bug?

### ISSUE THREAD

**Issue by scott0122** [slic3r/Slic3r#4406]
### Version
1.3.0-dev, Slic3r-master.2018.05.07.1676.8a93fbb.64bit

### Operating system type + version
Windows 10 64bit

### Behavior
My infill setting
![snag-0001](https://user-images.githubusercontent.com/39105828/39788795-b5976be2-535e-11e8-9327-f1a64cd22c93.png)
When I place model on this direction
![snag-0002](https://user-images.githubusercontent.com/39105828/39788806-c0b596fc-535e-11e8-8729-0f6669d3526c.png)
Infill on some layer seems broken
![snag-0003](https://user-images.githubusercontent.com/39105828/39788814-cbd2c424-535e-11e8-9cb9-95309d3943a2.png)
![snag-0004](https://user-images.githubusercontent.com/39105828/39788818-ce5b595e-535e-11e8-9719-ca96e2b80122.png)

When I place model on this direction, the problem is gone.
![snag-0005](https://user-images.githubusercontent.com/39105828/39788831-d9e7c0fa-535e-11e8-85a3-7898385495bd.png)
![snag-0006](https://user-images.githubusercontent.com/39105828/39788833-dc075f30-535e-11e8-8e82-5e79604ca482.png)
![snag-0007](https://user-images.githubusercontent.com/39105828/39788835-de246466-535e-11e8-9727-34d949cd3477.png)

#### STL/Config (.ZIP) where problem occurs
[Body1.zip](https://github.com/slic3r/Slic3r/files/1985977/Body1.zip)

**Comment by foreachthing** [comment]
@scott0122 I think you need to go over here: https://github.com/prusa3d/Slic3r/issues/

Also, see here: https://github.com/prusa3d/Slic3r/issues/824
And maybe this one, too: https://github.com/prusa3d/Slic3r/issues/825

**Comment by bubnikv** [comment]
Yes, it is  prusa3d#825 <https://github.com/prusa3d/Slic3r/issues/825> and
there is the fix included.

**Comment by foreachthing** [comment]
@bubnikv would you be kind enough to PR your changes to Slic3r as well? Thanks!

**Comment by supermerill** [comment]
I'm not at home this week. I can't test a slic3r build right now.
I should have PR the bubnikv changes, but as it's a bit time consuming, i waited the next release of slic3rPE. As it's now out, I promise to test it & PR as soon as possible. If someone want to Merge back the changes & fix before me, no problem.

**Comment by lordofhyphens** [comment]
@foreachthing that should be a separate Slic3r issue then.

**Comment by KillyMXI** [comment]
I experience this issue with gyroid pattern in Slic3r 1.3.0.
Unfortunately, with some of my models, 90deg rotation just makes the issue appear in a different part of a model.

**Comment by Yenya** [comment]
I see the same problem with this part:

https://github.com/gregsaun/bear_extruder_and_x_axis/blob/master/printed_parts/stl/mk3s/x_carriage.stl

0.2mm layers, 0.4mm nozzle, 4 solid top and bottom layers, 20 % gyroid infill, min 4 perimeters. After the last solid bottom layer there is a single layer with complete gyroid infill, but the layers above it up to about 2.4 mm have almost half of the infill missing.  Honeycomb works as expected.
![x_carriage](https://github.com/slic3r/Slic3r/assets/1055951/2e786e20-1eac-4e65-b1b1-3399e05d63cf)
